lib/src/view/game/analysis_screen.dart

```diff
@@ -0,0 +1,993 @@
+import 'dart:ui';
+import 'package:collection/collection.dart';
+import 'package:dartchess/dartchess.dart';
+import 'package:chessground/chessground.dart' as cg;
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lichess_mobile/src/constants.dart';
+import 'package:lichess_mobile/src/model/common/chess.dart';
+import 'package:lichess_mobile/src/model/common/eval.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/model/common/node.dart';
+import 'package:lichess_mobile/src/model/common/uci.dart';
+import 'package:lichess_mobile/src/model/engine/engine_evaluation.dart';
+import 'package:lichess_mobile/src/model/game/analysis_ctrl.dart';
+import 'package:lichess_mobile/src/model/settings/board_preferences.dart';
+import 'package:lichess_mobile/src/model/settings/general_preferences.dart';
+import 'package:lichess_mobile/src/styles/styles.dart';
+import 'package:lichess_mobile/src/utils/immersive_mode.dart';
+import 'package:lichess_mobile/src/utils/rate_limit.dart';
+import 'package:lichess_mobile/src/view/engine/engine_gauge.dart';
+import 'package:lichess_mobile/src/widgets/adaptive_action_sheet.dart';
+import 'package:lichess_mobile/src/widgets/adaptive_bottom_sheet.dart';
+import 'package:lichess_mobile/src/widgets/buttons.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+import 'package:lichess_mobile/src/widgets/non_linear_slider.dart';
+import 'package:lichess_mobile/src/widgets/platform.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/utils/chessground_compat.dart';
+import 'package:lichess_mobile/src/model/game/game.dart';
+import 'package:lichess_mobile/src/widgets/settings.dart';
+
+class AnalysisScreen extends ConsumerWidget {
+  const AnalysisScreen({
+    required this.variant,
+    required this.steps,
+    required this.orientation,
+    required this.id,
+  });
+
+  final Variant variant;
+  final IList<GameStep> steps;
+  final Side orientation;
+  final ID id;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return ConsumerPlatformWidget(
+      androidBuilder: _androidBuilder,
+      iosBuilder: _iosBuilder,
+      ref: ref,
+    );
+  }
+
+  Widget _androidBuilder(BuildContext context, WidgetRef ref) {
+    final ctrlProvider = analysisCtrlProvider(variant, steps, orientation, id);
+
+    final evalContext = ref.watch(
+      ctrlProvider.select((value) => value.evaluationContext),
+    );
+    final depth = ref.watch(
```

Comment:
This will cause the whole screen to rebuild. I think it's better to send `evalContext` to the Engine Depth widget to prevent that.